task-driver/src/settle_match.rs

```diff
@@ -302,85 +238,36 @@ impl SettleMatchTask {
             && tx_rejection.to_string().contains(NULLIFIER_USED_ERROR_MSG){
                 return Ok(())
             }
-        let tx_hash =
-            tx_submit_res.map_err(|err| SettleMatchTaskError::StarknetClient(err.to_string()))?;
-
-        log::info!("tx hash: 0x{:x}", starknet_felt_to_biguint(&tx_hash));
-        let status = self
-            .starknet_client
-            .poll_transaction_completed(tx_hash)
-            .await
-            .map_err(|err| SettleMatchTaskError::StarknetClient(err.to_string()))?;
-        if status == TransactionStatus::Reverted || status == TransactionStatus::Aborted {
-            return Err(SettleMatchTaskError::StarknetClient(format!(
-                "transaction failed with status: {:?}",
-                status
-            )));
-        }
-
-        // If the transaction was successful, cancel all orders on both nullifiers,
-        // await new validity proofs
-        self.global_state.nullify_orders(party0_reblind_proof.original_shares_nullifier).await;
-        self.global_state.nullify_orders(party1_reblind_proof.original_shares_nullifier).await;
 
-        Ok(())
+        tx_submit_res.map_err(|e| SettleMatchTaskError::Arbitrum(e.to_string()))
     }
 
     /// Apply the match result to the local wallet, find the wallet's new
     /// Merkle opening, and update the global state
     async fn update_wallet_state(&self) -> Result<(), SettleMatchTaskError> {
+        // Cancel all orders on both nullifiers, await new validity proofs
+        let party0_reblind_proof = &self.party0_validity_proof.reblind_proof.statement;
+        let party1_reblind_proof = &self.party1_validity_proof.reblind_proof.statement;
```

Comment:
tiny nit but can we name these `party_*_reblind_statement`? just reads a little strange to see `let *_proof = *.statement`